### PR TITLE
Remove unused arguments

### PR DESCRIPTION
**Pull request by fabianegli**
Remove unused arguments and some other minor code maintenance.

WIP until #1817 is clarified. => [It has been decided](https://nfcore.slack.com/archives/CE5LG7WMB/p1666340006328949?thread_ts=1666334956.682939&cid=CE5LG7WMB) to make these changes - WIP lablel removed.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/utils.py [hunk 1]
```diff
@@ -339,22 +339,22 @@ def setup_requests_cachedir():
...
-       poll_every (int): How many tenths of a second to wait between function calls. Default: 20.
+       refresh_per_second (int): Refresh this many times per second. Default: 20.
 
     Returns:
        None. Just sits in an infite loop until the function returns True.
     """
     try:
         spinner = Spinner("dots2", "Use ctrl+c to stop waiting and force exit.")
-        with Live(spinner, refresh_per_second=20):
+        with Live(spinner, refresh_per_second=refresh_per_second):
```
Comment (ewels) [inline nf_core/utils.py:357]: I'm not sure if we even need to parameterise this. It's how often the _spinner_ refreshes and the actual polling runs every 2 seconds with the hardcoded `sleep` call. I think we can just hardcode the 20 fps spinner.

**Review by ewels — APPROVED** [review]
Nice!

File: nf_core/utils.py (hunk 1 above)
Comment (fabianegli) [reply nf_core/utils.py:357]: But then why the argument in the function? If this is hardcoded the argument is mute - no?

Comment (fabianegli) [reply nf_core/utils.py:357]: https://github.com/nf-core/tools/blob/f3160d096368ae2b3118562276b5248011f59907/nf_core/utils.py#L342

**Pull request merged**